applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h

```diff
@@ -312,6 +316,147 @@ class GeometryUtilities
         return length;
     }
 
+// --------------------------------------------------------------------------
+    double ComputeVolume()
+    {
+        KRATOS_TRY
+
+        std::function<double(const GeometryType&)> volume_method;
+
+        const int domain_size = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+        const SizeType number_of_nodes = GetNumberOfElementalNodesInModelPart(mrModelPart);
+
+        if (domain_size == 2) {
+            if (number_of_nodes == 3) {
+                volume_method = [](const GeometryType& rGeometry) {
+                    return std::pow(
+                        ElementSizeCalculator<2, 3>::AverageElementSize(rGeometry), 2);
+                };
+            } else if (number_of_nodes == 4) {
+                volume_method = [](const GeometryType& rGeometry) {
+                    return std::pow(
+                        ElementSizeCalculator<2, 4>::AverageElementSize(rGeometry), 2);
+                };
+            } else {
+                KRATOS_ERROR
+                    << "Unsupported geometry type. Only geometries with 3 and "
+                       "4 nodes are supported in 2D. [ number_of_nodes = "
+                    << number_of_nodes << " ].\n";
+            }
+        } else if (domain_size == 3) {
+            if (number_of_nodes == 4) {
+                volume_method = [](const GeometryType& rGeometry) {
+                    return std::pow(
+                        ElementSizeCalculator<3, 4>::AverageElementSize(rGeometry), 3);
+                };
+            } else if (number_of_nodes == 6) {
+                volume_method = [](const GeometryType& rGeometry) {
+                    return std::pow(
+                        ElementSizeCalculator<3, 6>::AverageElementSize(rGeometry), 3);
+                };
+            } else if (number_of_nodes == 8) {
+                volume_method = [](const GeometryType& rGeometry) {
+                    return std::pow(
+                        ElementSizeCalculator<3, 8>::AverageElementSize(rGeometry), 3);
```

Comment:
For understanding:

What is the `AverageElementSize` of a single element? There is only one single element size - to my understanding. 
Why can you not directly call the `DomainSize` function at the element? The current implementation does not work for mixed element types, which i find very strange?

Also, why is the `AverageElementSize` value taken to the power of 3?

Reply:
`AverageElementSize` is the square root of area in 2D, and cubic root of volume in 3D. Thats why I am taking 3rd power in 3D case and 2nd power in 2D case. Once you raise the power, it is same as `DomainSize`. There is no other reason for me except to be consistent with function call names in volume calculation and its derivative calculation since we don't have an interface to calculate `DomainSize` derivative in `Geometry` class.

I will simplyfy the `ComputeVolume` calculation to use `DomainSize`, it will be much simpler in that case. Thanks :)